src/components/capture/Edit.tsx

```diff
@@ -156,42 +171,91 @@ function CaptureEdit() {
         },
     };
 
-    // const discoversSubscription = (discoverDraftId: string) => {
-    //     setDraftId(null);
-    //     return startSubscription(
-    //         supabaseClient.from(
-    //             `${TABLES.DISCOVERS}:draft_id=eq.${discoverDraftId}`
-    //         ),
-    //         (payload: any) => {
-    //             setDraftId(payload.draft_id);
-    //             setFormState({
-    //                 status: FormStatus.GENERATED,
-    //             });
-    //             trackEvent(payload);
-    //         },
-    //         (payload: any) => {
-    //             helpers.jobFailed(`${messagePrefix}.test.failedErrorTitle`);
-    //             trackEvent(payload);
-    //         }
-    //     );
-    // };
+    const storeUpdatedDraftSpec = async () => {
+        // TODO (optimization | typing): Narrow the columns selected from the draft_specs_ext table.
+        //   More columns are selected than required to appease the typing of the editor store.
+        const draftSpecsResponse = await supabaseClient
+            .from(TABLES.DRAFT_SPECS_EXT)
+            .select(`catalog_name,draft_id,expect_pub_id,spec,spec_type`)
+            .eq('draft_id', editDraftId)
+            .eq('spec_type', ENTITY.CAPTURE)
+            .then(handleSuccess<DraftSpecQuery[]>, handleFailure);
+        if (draftSpecsResponse.error) {
+            return helpers.callFailed({
+                error: {
+                    title: 'captureCreate.generate.failedErrorTitle',
+                    error: draftSpecsResponse.error,
+                },
+            });
+        }
+
+        if (draftSpecsResponse.data && draftSpecsResponse.data.length > 0) {
+            setDraftSpecs(draftSpecsResponse.data);
+
+            void mutateDraftSpecs();
+        }
+    };
+
+    const discoversSubscription = (discoverDraftId: string) => {
+        setDraftId(null);
+
+        jobStatusPoller(
+            supabaseClient
+                .from(TABLES.DISCOVERS)
+                .select(
+                    `
+                    draft_id,
+                    job_status
+                `
+                )
+                .match({
+                    draft_id: discoverDraftId,
+                }),
+            (payload: any) => {
```

Comment:
Polling only occurs on the initial discovery for edit since a new draft is not created when the generate button is clicked. However, a call to mutate should be made after the draft specs table is update via the discovery RPC and this polling function is the only insight into when the job is successful. One could listen to discovers entry with the latest creation date to remove the noise. While I am not a fan of being reliant on timestamps, it sounds like the best move in this stream of consciousness. Will revisit this tomorrow (Oct. 11th).